src/analysis/credResult.js

```diff
@@ -0,0 +1,87 @@
+// @flow
+
+import {
+  type WeightedGraph,
+  type WeightedGraphJSON,
+  toJSON as wgToJSON,
+  fromJSON as wgFromJSON,
+} from "../core/weightedGraph";
+import {type Compatible, toCompat, fromCompat} from "../util/compat";
+import {
+  type TimelineCredParameters,
+  type TimelineCredParametersJSON,
+  paramsToJSON,
+  paramsFromJSON,
+} from "./timeline/params";
+import {
+  type PluginDeclarations,
+  type PluginDeclarationsJSON,
+  toJSON as pluginsToJSON,
+  fromJSON as pluginsFromJSON,
+} from "./pluginDeclaration";
+import {timelinePagerank} from "../core/algorithm/timelinePagerank";
+import {type CredData, computeCredData} from "./credData";
+import {distributionToCred} from "../core/algorithm/distributionToCred";
+
+/**
+ * Comprehensive cred output data, including the graph, the scores, the params, and the plugins.
+ */
+export type CredResult = {|
+  // The Graph on which this cred data was computed.
+  +weightedGraph: WeightedGraph,
+  // Cred level information, always stored in graph address order.
+  +credData: CredData,
+  // Parameters used to compute this cred data
+  +params: TimelineCredParameters,
+  // Plugin declarations used to compute this cred data
+  +plugins: PluginDeclarations,
+|};
+
+export async function compute(
+  wg: WeightedGraph,
+  params: TimelineCredParameters,
+  plugins: PluginDeclarations
+): Promise<CredResult> {
+  const {graph} = wg;
+  const nodeOrder = Array.from(graph.nodes()).map((x) => x.address);
+  const userTypes = [].concat(...plugins.map((x) => x.userTypes));
+  const scorePrefixes = userTypes.map((x) => x.prefix);
+  const distribution = await timelinePagerank(
+    wg,
+    params.intervalDecay,
+    params.alpha
+  );
+  const credScores = distributionToCred(distribution, nodeOrder, scorePrefixes);
+  const credData = computeCredData(credScores);
+  return {weightedGraph: wg, credData, params, plugins};
+}
+
+const COMPAT_INFO = {type: "sourcecred/credResult", version: "0.1.0"};
+
+export type CredResultJSON = Compatible<{|
+  +weightedGraph: WeightedGraphJSON,
+  +credData: CredData,
+  +params: TimelineCredParametersJSON,
+  +plugins: PluginDeclarationsJSON,
+|}>;
+
+export function toJSON(x: CredResult): CredResultJSON {
+  const {weightedGraph, credData, params, plugins} = x;
+  return toCompat(COMPAT_INFO, {
+    weightedGraph: wgToJSON(weightedGraph),
+    credData,
```

Comment:
Note: Currently the credData object is just a raw JSON object, so we don't need any special serialization here.